### PR TITLE
Add blocks_not_found in RPC blocks_info response rather than an error

### PR DESCRIPTION
**Pull request by guilhermelawless**
Wiki markup:

**INCOMPLETE** - needs to be updated with latest changes

---

## Retrieve multiple blocks with additional info

Retrieves a json representations of **blocks** with block **account**, transaction **amount**, block **balance**, block **height** in account chain, block local modification **timstamp**, block **confirmation status** (since version 19.0), **subtype** (for state blocks since version 19.0).

Starting from version 19.0, **blocks_not_found** is an additional field in the response with the blocks that were not found locally. Before this version, an error would occur instead.

Request
```json
{
  "action": "blocks_info",
  "hashes": ["87434F8041869A01C8F6F263B87972D7BA443A72E0A97D7A3FD0CCC2358FD6F9", "0000000000000000000000000000000000000000000000000000000000000001"]
}
```
Response:
```json
{
  "blocks" : {
    "87434F8041869A01C8F6F263B87972D7BA443A72E0A97D7A3FD0CCC2358FD6F9": {
         "block_account": "xrb_1ipx847tk8o46pwxt5qjdbncjqcbwcc1rrmqnkztrfjy5k7z4imsrata9est",
         "amount": "30000000000000000000000000000000000",
         "balance": "5606157000000000000000000000000000000",
         "height": "58",
         "local_timestamp": "0",
         "confirmed": "false",
       "contents": "{\n
         \"type\": \"state\",\n
         \"account\": \"xrb_1ipx847tk8o46pwxt5qjdbncjqcbwcc1rrmqnkztrfjy5k7z4imsrata9est\",\n
         \"previous\": \"CE898C131AAEE25E05362F247760F8A3ACF34A9796A5AE0D9204E86B0637965E\",\n
         \"representative\": \"xrb_1stofnrxuz3cai7ze75o174bpm7scwj9jn3nxsn8ntzg784jf1gzn1jjdkou\",\n
         \"balance\": \"5606157000000000000000000000000000000\",\n
         \"link\": \"5D1AA8A45F8736519D707FCB375976A7F9AF795091021D7E9C7548D6F45DD8D5\",\n
         \"link_as_account\": \"xrb_1qato4k7z3spc8gq1zyd8xeqfbzsoxwo36a45ozbrxcatut7up8ohyardu1z\",\n
         \"signature\": \"82D41BC16F313E4B2243D14DFFA2FB04679C540C2095FEE7EAE0F2F26880AD56DD48D87A7CC5DD760C5B2D76EE2C205506AA557BF00B60D8DEE312EC7343A501\",\n
         \"work\": \"8a142e07a10996d5\"\n
      }\n",
      "subtype": "send"
     }
  },
  "blocks_not_found": [
    "0000000000000000000000000000000000000000000000000000000000000001"
  ]
}
```

(...)

### CODE REVIEW

**Comment by guilhermelawless** [comment]
`blocks_not_found` is always included in the response even if there are no blocks there. Could also not include it if it's empty. I think any of the approaches are reasonable.

**Comment by wezrule** [comment]
This is a breaking change if someone expects an error if any of the blocks can't be found; this is the pattern we have in quite a few places. There should probably be an optional `bool` argument to turn this behavior on/off so that it's backwards compatible. What do you think @zhyatt?

**Comment by guilhermelawless** [comment]
It's a good point, though I doubt someone is using this to find out the blocks can't be found, as it doesn't offer any feedback regarding which. Might be worth breaking for added convenience.

**Comment by guilhermelawless** [comment]
It's a good point. Could also include the `error` field if someone is relying on that, along with the rest of fields.

**Comment by zhyatt** [comment]
I imagine the current behavior is more of an issue than the proposed fix would cause, but we don't want to do breaking changes now. I am not sure including `error` when there is a partially valid response makes sense, do we use that pattern anywhere else? I would lean toward an optional argument to enable this behavior of providing `blocks_not_found` in the response, and then when we do a bigger refactor RPC we can evaluate removing that and going to a standard `blocks_not_found` response because breaking changes for RPC will be happening then anyways.

**Comment by guilhermelawless** [comment]
@wezrule Do you use the `bool` to change the type of response anywhere else? Would like to follow the naming convention.

**Comment by wezrule** [comment]
> @wezrule Do you use the `bool` to change the type of response anywhere else? Would like to follow the naming convention.

Look like `output_raw` is an example inside `void nano::json_handler::account_history ()` (`json_handler.cpp`).

File: nano/rpc_test/rpc.cpp [hunk 1]
```diff
@@ -4423,54 +4423,89 @@ TEST (rpc, blocks_info)
 	nano::ipc_rpc_processor ipc_rpc_processor (system.io_ctx, rpc_config);
 	nano::rpc rpc (system.io_ctx, rpc_config, ipc_rpc_processor);
 	rpc.start ();
+	auto check_blocks = [&](test_response & response) {
```
Comment (wezrule) [inline nano/rpc_test/rpc.cpp]: Can you pass in exactly what you need in this lambda capture. Otherwise LGTM.

**Pull request merged**